spectrocloud/cluster_common_virtual_machine.go

```diff
@@ -280,6 +292,87 @@ func toSpecCreateRequest(d *schema.ResourceData) *models.V1ClusterVirtualMachine
 	return vmSpec
 }
 
+func toDataVolumeTemplates(d *schema.ResourceData) []*models.V1VMDataVolumeTemplateSpec {
+	volumeSpec := d.Get("volume_spec").(*schema.Set)
+	var dataVolumeTemplates []*models.V1VMDataVolumeTemplateSpec
+
+	if volumeSpec != nil {
+		volumeSpecList := volumeSpec.List()
+		for _, volume := range volumeSpecList {
+			volumeMap := volume.(map[string]interface{})
+			if volumeData, ok := volumeMap["volume"]; ok {
+				volumeDataList := volumeData.([]interface{})
+				for _, dataVolume := range volumeDataList {
+					dataVolumeMap := dataVolume.(map[string]interface{})
+					if _, ok := dataVolumeMap["data_volume"]; ok && len(dataVolumeMap["data_volume"].(*schema.Set).List()) > 0 {
+						dataVolumeTemplates = append(dataVolumeTemplates, toDataVolumeTemplateSpecCreateRequest(dataVolumeMap["data_volume"], dataVolumeMap["name"].(string), d.Get("name").(string)))
+					}
+				}
+			}
+		}
+	}
+	return dataVolumeTemplates
+}
+
+func toDataVolumeTemplateSpecCreateRequest(dataVolumeSet interface{}, name string, vmname string) *models.V1VMDataVolumeTemplateSpec {
+	dataVolumeList := dataVolumeSet.(*schema.Set).List()
+
+	for _, dataVolume := range dataVolumeList {
+		volume := dataVolume.(map[string]interface{})
+		storage := volume["storage"].(string)
+
+		dataVolumeTemplate := &models.V1VMDataVolumeTemplateSpec{
+			Metadata: &models.V1VMObjectMeta{
+				OwnerReferences: []*models.V1VMOwnerReference{
+					{
+						APIVersion: ptr.StringPtr("kubevirt.io/v1"),
+						Kind:       ptr.StringPtr("VirtualMachine"),
+						Name:       ptr.StringPtr(vmname),
+						UID:        ptr.StringPtr(""),
+					},
+				},
+				Name: "disk-0-vol",
+			},
+			Spec: &models.V1VMDataVolumeSpec{
+				//Storage: toV1VMStorageSpec(storage),
+				Pvc: toV1VMPersistentVolumeClaimSpec(storage),
+				Source: &models.V1VMDataVolumeSource{
+					Blank: make(map[string]interface{}),
+				},
+			},
+		}
+
+		return dataVolumeTemplate
+	}
+
+	return &models.V1VMDataVolumeTemplateSpec{}
+}
+
+func toV1VMPersistentVolumeClaimSpec(storage string) *models.V1VMPersistentVolumeClaimSpec {
+	return &models.V1VMPersistentVolumeClaimSpec{
+		Resources: &models.V1VMCoreResourceRequirements{
+			Requests: map[string]models.V1VMQuantity{
+				"storage": models.V1VMQuantity(storage),
+			},
+		},
+		StorageClassName: "sumit-storage-class",
```

Comment:
This should be dynamic

Reply:
(with `spectro-storage-class` as the default)

Reply:
@TylerGillson thank you for review! We have decided to take another turn and now this whole file is gone. I addressed this in another PR. Let's get back to review in the middle of this week.